nautobot/extras/factory.py

```diff
@@ -283,7 +284,7 @@ def _value(self):
         ):
             return faker.Faker().pystr()
         elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_JSON:
-            return faker.Faker().pydict(allowed_types=[str])
+            return json.dumps(faker.Faker().pydict(allowed_types=[str]))
```

Comment:
This feels wrong to me? We *should* be able to store a dict in JSON metadata, not just a JSON string.

Reply:
Yeah the problem lies within `test_recreate_object_csv` again. When it recreates the data, it converts itself into a JSON string instead of a `dict` object:
```
..s...................ss...F..
======================================================================
FAIL: test_recreate_object_csv (nautobot.extras.tests.test_api.ObjectMetadataTest.test_recreate_object_csv)
CSV export an object, delete it, and recreate it via CSV import.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/source/nautobot/core/testing/api.py", line 725, in test_recreate_object_csv
    self.assertEqual(
AssertionError: {'bit': 'zPeGIyqEgYhnQApyvxIq', 'lose': '[351 chars]HiJ'} != '{"bit": "zPeGIyqEgYhnQApyvxIq", "lose": [353 chars]iJ"}' : value should have been unchanged on delete/recreate but it differs!
```